### ISSUE TITLE
World loading makes damaged items permanently lose one point of durability and are impossible to reinforce from then on

### ISSUE THREAD

**Issue by Rizoel95** [CleverRaven/Cataclysm-DDA#18315]
### Scroll down about halfway for the TL;DR and then a bit more to learn about how to prevent this from happening to items.

As the title says. Saving and quitting with damaged items in the world (or in your inventory), will make those items permanently lose one point of durability, and will also make them impossible to reinforce once repaired, **although the game will say they are reinforced**.

**Example:**
Let's say you just killed a zombie, and he dropped a slightly damaged ( or '**|\**' ) t-shirt (filthiness has nothing to do with this; the following bug happens to all clothing regardless of cleanliness, and to every other item in the game for that matter; guns, weapons, tools, etc). You pick up the **|\** t-shirt to repair and reinforce later once you gain some tailoring skill. Now let's say you save and quit the game right then and there. Congratulations; that t-shirt, and **all other damaged** items in the world and in your inventory will now be impossible to reinforce. Forever.

But let's say you want to try and reinforce it anyway; you have good tailoring skills now. The first thing you notice, is that the t-shirt is now **|.** instead of **|\**. This always happens upon saving and quitting the game; saving and quitting causes **damaged** items to lose an extra point of durability. Fortunately, if this does happen to an item, it will never happen to it again, but this is also **permanent**, which I'll explain in a second.

You repair the **|.** t-shirt to **|\** as normal. You repair it again, and suddenly the game says that it's reinforced (**++**); it apparently skipped **||** and went straight to **++**.

'Oh cool', you think, 'slightly damaged items will always be reinforced upon repairing, awesome.'

Except no.

See, the t-shirt is **not** actually reinforced; the t-shirt actually **didn't** skip its undamaged state ( **||** ) when repairing, and it is indeed actually a **||** item, and will act like a **||** item. and will not have stats that a reinforced t-shirt would have. It **IS** a **||** item, but the game will insist on saying it's reinforced. The game will even skip the refitting step to do this 'fake reinforcement' if the item in question is clothing, which you normally cannot do; you need to refit the thing first before you can reinforce it, if the clothing can be refitted. The 'fake reinforced' t-shirt can however still be refitted if this happens.

So now you walk around with your apparently '**++**' t-shirt. You turn a corner and get clawed in the chest by a zombie, ripping your **++** t-shirt. You kill the zombie, and take a look at your t-shirt. It should be **||** now, right?

It is actually **|\** instead.

This is because this item was, again, never actually reinforced. 'Fake reinforced' items will be taken to **|\** when damaged, because they are actually **||**.

These items, once bugged, cannot ever be reinforced for real.

Fortunately, there is a way to prevent this from happening to items; when you save and quit, undamaged ( **||** ) and reinforced ( **++** ) items will not have this bug happen to them, provided they don't ever experience saving and quitting while they are damaged. Items that have never experienced the world saving and quitting while damaged, will not experience this bug and will behave completely normal, as long as they are that way when saving and quiting.

So basically, don't save and quit while the item is **|\** or below, and the item will be fine. If you **do** save and quit while it's damaged, it's eff'ed forever.

And for those of us that use quicksave all the time; if you have an undamaged ( **||** ) and unbugged item, and it gets damaged ( **|\** or below ), and you quicksave the game (so, no quitting happening), but then repair it, the item will still be fine if you quicksave after that or when you save and quit for real.

In fact, that right there makes this bug sound like it has something to do with the game **loading** up items, and actually nothing to do with saving. But I'll let you experts find out what's actually the cause behind this thing.

The only workaround I've figured out to fix bugged items, is to spawn an undamaged version of that item with the debug menu, and keep that in tip-top shape, and make sure you at least repair it to normal when it gets damaged before you save and quit. (Speaking of the debug menu, we really need a way to set damage on items; I've bashed enough t-shirts against walls for one day I feel)
## TL;DR: Loading up a world causes ALL damaged items in the world to be bugged and inreinforcable forever and the game will lie that they are reinforced.

---

**To Reproduce:**

1) Spawn some repairing equipment (either sewing kit, thread, rags, AND/OR soldering iron, batteries, scrap, etc.)

2) Give character repairing skills (about 6 Tailoring and/or 8 mechanics)

3) Spawn something fragile (T-shirt will do) because we're gonna smack it against a wall to damage it, unless you've found a better way to damage stuff.

4) Smack item against the wall until it's **|\**.

5) Save and quit, re-load save, item should be **|.** instead. It will now be bugged.

6) Repair the item to full. The game will apparently skip **||** and go straight to **++**. The item **is** actually **||** though, and will fuction as one, and will not let you reinforce it for real.

---

**How to prevent this from happening to items and how to fix items affected by all this:**

Make sure the item in question has never experienced the world 'saving and quitting' while it is |\ or lower, otherwise it will be bugged forever.

If the item has been bugged, debug spawn another one of that item, and from then on make sure to never save and quit while it is damaged ( **|\** or lower). **Quicksaving** while it's damaged is fine, quitting the world so it has to load a damaged item isn't.

Basically, if the world needs to load a damaged item, that item will be bugged.

If a damaged item is found in the world (say, because a zombie drops it), repair it before saving and quitting.

---

All testing was done on the latest experimental as of the time of this writing.

---

Btw, quick question regarding another bug I think I've found (I'm not quite sure if it's actually a bug or intended, and will make separate report for it if it isn't).

Clothing (and this one is specifically affecting clothing; weapons and everything else seem to be fine when 'real reinforced') no longer receive a slight buff to protection upon being reinforced. As in, for example, reinforcing a winter coat no longer seems to give it a few extra points in bash and cutting protection. And I'm not talking about the special tailoring kit that lets you line clothing with leather and such, I'm talking about just regular reinforcing.

Is this intended to not happen anymore? Because I swear reinforced clothing used to do this.

**Comment by Beanzy** [comment]
So that's what it was. I thought I saw `|\` items turn into `|.` items when I encountered them later on, I just chalked it up to poor memory on my part.

On another note, I noticed that sometimes a `|.` item would repair into a `\.` item (and I'm sure it's not because my character botched the repair) that `\.` then could be repaired into a `|.` item; I wonder if this could be related to the item being more damaged than the game says it is?

**Comment by Beanzy** [comment]
I've provided an example of the bug I described in the screenshot below.

![cdda bug](https://cloud.githubusercontent.com/assets/15885173/18495075/e6329b60-79e8-11e6-84b3-b04c7899a9c7.PNG)

In the above, I've sorted the items by 'damage' in the advanced inventory screen. As you can see, there are `\.` items listed below `..` items on the screen.

If you were to try to repair one of these `\.` items, you'd get a `..` damaged item after the first successful repair, which could then be repaired into an actual `\.` durability item.

I'm going to guess that this is the result of a `..` durability item being damaged as a result of the bug Rizoel95 described.

**Comment by Zireael07** [comment]
What version are you on? I'm wearing a '|\' item of clothing and I've loaded and saved twice, not seeing the issue in g5a5f641

**Comment by Beanzy** [comment]
Jenkins build 5498 from 9/14/16.

**Comment by Zireael07** [comment]
My build is 5507 from 9/16/16. Try updating? Does the issue persist?

**Comment by Beanzy** [comment]
Just tested on build 5525 from today, the bug is still in the game.

I'm using both static and dynamic npcs, static zombies, no mods. I saved and then closed C:DDA when testing the bug.

**Comment by Zireael07** [comment]
Weird. No one else is reporting this. What platform are you on? Did you build from source or are you using narc's binaries?

**Comment by Beanzy** [comment]
I'm using the officially provided Jenkins build (http://dev.narc.ro/cataclysm/jenkins-latest/Windows_x64/Tiles/)

Did you test this on a new world? It could very well be related to world gen settings. 

Steps to reproduce:
1) Download a build from provided link (I used 5542)

2) Extract build into new folder

3) Start game and create world with default settings

4) Create character

5) Once ingame find damaged items (I just debugged a weapon and killed a zombie)

6) Hit "shift + s" to save and quit

7) Exit Cataclysm DDA

8) Start C:DDA and load save

9) Note that damaged items have now taken one tick of damage

**Comment by Coolthulhu** [comment]
I can't reproduce this.
Could be fixed already.

Leaving it for now with both 0.D and "Confirmation Needed" tags - if it is still there, it is serious.

**Comment by mugling** [comment]
This should have been closed by recent updates to JSON save/load code. Can be reopened if reproducible in an new save (please attach)
